modules/data.atmosphere/R/spin.met.R

```diff
@@ -51,25 +52,43 @@ spin.met <- function(in.path,in.prefix,start_date,end_date,nyear,nsample,resampl
   
   # spin settings
   if(missing(nyear)|is.null(nyear) | is.na(nyear)) nyear <- 1000
+  nyear <- as.numeric(nyear)
   if(missing(nsample)|is.null(nsample) | is.na(nsample)) nsample <- 50
+  nsample <- as.numeric(nsample)
   nsample <- min(nsample,length(avail.years))
+  avail.years <- avail.years[seq_len(nsample)]
   if(missing(resample) | is.null(resample)|is.na(resample)) resample <- TRUE
-  spin_start_date <- as.POSIXct(start_date,"UTC") - lubridate::years(nyear)
+  resample <- as.logical(resample)
+  spin_start_date <- as.POSIXct(run_start_date,"UTC") - lubridate::years(nyear)
   
   ### define the met years to sample
+  new_year <- seq(lubridate::year(spin_start_date),by=1,length.out=nyear)
+  is.leap <- lubridate::leap_year(avail.years)
+  spin_year <- NA
   if(resample){
-    spin_year <- sample(avail.years[1:nsample],
-                        size = nyear,replace = TRUE)
+    for(t in seq_along(new_year)){
+      if(lubridate::leap_year(new_year[t])){
```

Comment:
I didn't do this by choice (or with foresight), but in response to a bug -- models that need leap years throw an error if you pass them met that's one day short